lib/usermanager.js

```diff
@@ -293,14 +293,27 @@ exports = module.exports = {
     // only execute if we have a single matching record
     this.retrieveUser(search, function (error, result) {
       if (error) {
-        callback(error);
-      } else if (result) {
-        database.getDatabase(function(err,db){
-          db.update('user', search, update, callback);
-        }, configuration.getConfig('dbName'));
-      } else {
-        callback(new Error('No matching user record found'));
+        return callback(error);
       }
+      if (!result) {
+        return callback(new Error('No matching user record found'));
+      }
+      database.getDatabase(function(err, db) {
+        if (update.email_prev && update.email !== update.email_prev) {
```

Comment:
`update.email_prev` never seems to be defined for me and so the duplicate user error is not caught:

![error](https://user-images.githubusercontent.com/922987/52963132-e87b0c80-3396-11e9-8477-d6d7f97a79b3.png)